src/base/static/components/organisms/main-map.js

```diff
@@ -0,0 +1,445 @@
+        });
+      },
+      layersStatus: this.props.layersStatus,
+    });
+    this._map.on({
+      event: "layer:error",
+      callback: sourceId => {
+        if (sourceId === "composite") {
+          [sourceId] = this.lookupMapboxStyleId();
+        }
+        this.props.setLayerStatus(sourceId, {
+          status: "error",
+        });
+      },
+    });
+
+    // Handlers for map interaction events.
+    this._map.on({
+      event: "dragend",
+      callback: () => {
+        Util.log(
+          "USER",
+          "map",
+          "drag",
+          this._map.getBBoxString(),
+          this._map.getZoom(),
+        );
+        this.props.onDragend();
+      },
+    });
+    this._map.on({
+      event: "zoomend",
+      callback: () => {
+        Util.log("APP", "zoom", this._map.getZoom());
+        this.props.onZoomend();
+      },
+    });
+    this._map.on({ event: "movestart", callback: this.props.onMovestart });
+    this._map.on({
+      event: "moveend",
+      callback: () => {
+        Util.log("APP", "center-lat", this._map.getCenter().lat);
+        Util.log("APP", "center-lng", this._map.getCenter().lng);
+        this.props.setMapPosition({
+          center: this._map.getCenter(),
+          zoom: this._map.getZoom(),
+        });
+        this.props.onMoveend();
+      },
+    });
+
+    // Handlers for Mapseed place collections.
+    emitter.addListener("place-collection:loaded", layerId => {
+      this.props.setLayerStatus(layerId, {
+        status: "loaded",
+        type: "place",
+        isBasemap: false,
+        isVisible: !!this._layers.find(layer => layer.id === layerId)
+          .visible_default,
+      });
+    });
+    emitter.addListener("place-collection:add-place", collectionId => {
+      this.updatePlaceCollectionLayer(collectionId);
+    });
+    emitter.addListener(
+      "place-collection:focus-place",
+      ({ collectionId, modelId }) => {
+        this.updatePlaceCollectionLayer(collectionId, modelId);
+      },
+    );
+    emitter.addListener("place-collection:unfocus-all-places", () => {
+      Object.keys(this._places).forEach(collectionId => {
+        this.props.layersStatus[collectionId].isVisible &&
+          this.updatePlaceCollectionLayer(collectionId);
+      });
+    });
+
+    // Handlers for story-driven map transitions.
+    emitter.addListener(
+      constants.MAP_TRANSITION_FIT_LINESTRING_COORDS,
+      ({ coordinates }) => {
+        this._map.fitLineStringCoords(coordinates, { padding: 30 });
+      },
+    );
+    emitter.addListener(
+      constants.MAP_TRANSITION_FIT_POLYGON_COORDS,
+      ({ coordinates }) => {
+        this._map.fitPolygonCoords(coordinates, { padding: 30 });
+      },
+    );
+    emitter.addListener(
+      constants.MAP_TRANSITION_EASE_TO_POINT,
+      ({ coordinates, zoom }) => {
+        this._map.easeTo({
+          center: coordinates,
+          zoom: zoom,
+        });
+      },
+    );
+  }
+
+  lookupMapboxStyleId() {
```

Comment:
I think this can be a selector that lives in our Map duck.

Reply:
👍